course/src/main/java/org/openedx/course/presentation/container/CourseContainerFragment.kt

```diff
@@ -3,50 +3,99 @@ package org.openedx.course.presentation.container
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.ViewCompositionStrategy
 import androidx.core.os.bundleOf
 import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
-import androidx.viewpager2.widget.ViewPager2
+import androidx.lifecycle.lifecycleScope
 import com.google.android.material.snackbar.Snackbar
-import com.google.android.material.tabs.TabLayout
-import com.google.android.material.tabs.TabLayoutMediator
+import kotlinx.coroutines.launch
 import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import org.koin.core.parameter.parametersOf
+import org.openedx.core.FragmentViewType
 import org.openedx.core.extension.takeIfNotEmpty
 import org.openedx.core.presentation.global.viewBinding
+import org.openedx.core.ui.rememberWindowSize
 import org.openedx.core.ui.theme.OpenEdXTheme
 import org.openedx.course.R
 import org.openedx.course.databinding.FragmentCourseContainerBinding
 import org.openedx.course.presentation.CourseRouter
 import org.openedx.course.presentation.calendarsync.CalendarSyncDialog
 import org.openedx.course.presentation.calendarsync.CalendarSyncDialogType
-import org.openedx.course.presentation.container.CourseContainerTab
-import org.openedx.course.presentation.dates.CourseDatesFragment
-import org.openedx.course.presentation.handouts.HandoutsFragment
-import org.openedx.course.presentation.outline.CourseOutlineFragment
-import org.openedx.course.presentation.ui.CourseToolbar
-import org.openedx.course.presentation.videos.CourseVideosFragment
-import org.openedx.discussion.presentation.topics.DiscussionTopicsFragment
-import org.openedx.course.presentation.container.CourseContainerTab as Tabs
+import org.openedx.course.presentation.dates.CourseDatesScreen
+import org.openedx.course.presentation.dates.CourseDatesViewModel
+import org.openedx.course.presentation.handouts.HandoutsScreen
+import org.openedx.course.presentation.handouts.HandoutsType
+import org.openedx.course.presentation.outline.CourseOutlineScreen
+import org.openedx.course.presentation.outline.CourseOutlineViewModel
+import org.openedx.course.presentation.ui.CourseVideosScreen
+import org.openedx.course.presentation.videos.CourseVideoViewModel
+import org.openedx.discussion.presentation.DiscussionRouter
+import org.openedx.discussion.presentation.topics.DiscussionTopicsScreen
+import org.openedx.discussion.presentation.topics.DiscussionTopicsUIState
+import org.openedx.discussion.presentation.topics.DiscussionTopicsViewModel
 
 class CourseContainerFragment : Fragment(R.layout.fragment_course_container) {
 
     private val binding by viewBinding(FragmentCourseContainerBinding::bind)
+
+    private val courseRouter by inject<CourseRouter>()
+    private val discussionRouter by inject<DiscussionRouter>()
+
     private val viewModel by viewModel<CourseContainerViewModel> {
```

Comment:
5 view models inside one fragment! why are we doing this?

Reply:
I agree that it's not a better solution. I would like to use koinViewModel in compose functions but we need to control pull-refresh from CourseContainerFragment because collapsing header doesn't work correctly if we use pull-refresh on each screen. If you have suggestions, we can discuss them.

Reply:
Maybe it is possible to use effects based on SharedFlow, to control Fragment from viewModel? Also maybe we can try to inject viewModel directly inside composable?

Reply:
Yes, you are right. I added a solution based on events in the last commit. Please review it.